src/index.tsx

```diff
@@ -23,15 +25,17 @@ type CacheEntry = {
 
 export class ImageCache {
 
-    private getPath(uri: string, immutable?: boolean): string {
-        let path = uri.substring(uri.lastIndexOf("/"));
-        path = path.indexOf("?") === -1 ? path : path.substring(path.lastIndexOf("."), path.indexOf("?"));
-        const ext = path.indexOf(".") === -1 ? ".jpg" : path.substring(path.indexOf("."));
-        if (immutable === true) {
-            return BASE_DIR + "/" + SHA1(uri) + ext;
-        } else {
-            return BASE_DIR + "/" + s4() + s4() + "-" + s4() + "-" + s4() + "-" + s4() + "-" + s4() + s4() + s4() + ext;
-        }
+    private getPath(dbPath: string, immutable?: boolean): string {
+      let path = dbPath.substring(dbPath.lastIndexOf("/"));
+      const terms = dbPath.split("/");
+      const toEncode = terms[terms.length-2];
```

Comment:
room1/[dbkey]/image.jpg. 
Hence dbkey is unique, and potentially I dont need to encode this either.